lib/reconcile/utils.go

```diff
@@ -378,6 +379,86 @@ func getAstarteCommonEnvVars(deploymentName string, cr *apiv1alpha1.Astarte, bac
 	return appendRabbitMQConnectionEnvVars(ret, rpcPrefix+"RPC_AMQP_CONNECTION", cr)
 }
 
+func appendCassandraConnectionEnvVars(ret []v1.EnvVar, cr *apiv1alpha1.Astarte) []v1.EnvVar {
+	spec := cr.Spec.Cassandra.Connection
+	if spec != nil {
+		// pool size
+		if spec.PoolSize != nil {
+			ret = append(ret,
+				v1.EnvVar{
+					Name:  "CASSANDRA_POOL_SIZE",
+					Value: strconv.Itoa(*spec.PoolSize),
+				},
+			)
+		}
+
+		// autodiscovery
+		if pointy.BoolValue(spec.Autodiscovery, true) {
+			ret = append(ret,
+				v1.EnvVar{
+					Name:  "CASSANDRA_AUTODISCOVERY_ENABLED",
+					Value: "true",
+				},
+			)
+		}
+
+		// SSL
+		if spec.SSLConfiguration.Enabled {
+			ret = append(ret, v1.EnvVar{
+				Name:  "CASSANDRA_SSL_ENABLED",
+				Value: "true",
+			})
+
+			// CA configuration
+			if spec.SSLConfiguration.CustomCASecret.Name != "" {
+				// getAstarteCommonVolumes will mount the volume for us, if we're here. So trust the rest of our code.
+				ret = append(ret, v1.EnvVar{
+					Name:  "CASSANDRA_SSL_CA_FILE",
+					Value: "/cassandra-ssl/ca.crt",
+				})
+			}
+
+			// SNI configuration
+			switch {
+			case spec.SSLConfiguration.CustomSNI != "":
+				ret = append(ret, v1.EnvVar{
+					Name:  "CASSANDRA_SSL_CUSTOM_SNI",
+					Value: spec.SSLConfiguration.CustomSNI,
+				})
+			case !pointy.BoolValue(spec.SSLConfiguration.SNI, true):
+				ret = append(ret, v1.EnvVar{
+					Name:  "CASSANDRA_SSL_DISABLE_SNI",
+					Value: "true",
+				})
+			}
+		}
+
+		if spec.Secret != nil {
```

Comment:
This part is conceptually wrong, as you should enable this codepath if one between secret and username/password is entered.